sw/device/examples/sram_program/BUILD

```diff
@@ -42,23 +42,55 @@ opentitan_gdb_fpga_cw310_test(
     exit_success_pattern = "sram_program\\.c:47\\] PC: 0x1000208c, SRAM: \\[0x10000000, 0x10020000\\)",
     gdb_script = """
         target extended-remote :3333
-        set pagination off
 
         echo :::: Send OpenOCD the 'reset halt' command.\\n
         monitor reset halt
 
-        echo :::: Set RAM as RWX by executing code between _start and test_rom_main()\\n
-        file test_rom.elf
-        break rom_test_main
+        # Disable the watchdog timer as quickly as possible. If it fires, the
+        # exception will steal control from the debugger. If we're unlucky, the
+        # watchdog timer might fire before the commands below have disabled it,
+        # causing this test to fail. The proper solution would be to disable ROM
+        # execution by zeroing out CREATOR_SW_CFG_ROM_EXEC_EN in the OTP.
+        file rom.elf
+        break sec_mmio_init
         continue
+        finish
+        print watchdog_disable()
```

Comment:
Is there a particular reason for breaking `sec_mmio_init()` and executing until the end? I would actually prefer breaking earlier and using as little or no ROM code here if possible. It would be nice if we could break [here](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom/rom_start.S;l=139?q=rom_start.s&ss=opentitan%2Fopentitan). Then this test would cover the cases when the ROM execution is disabled but we enable it temporarily by writing to the `t0` register.

For disabling the watchdog timer, we can directly write to the memory-mapped register at `TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR + AON_TIMER_WDOG_CTRL_REG_OFFSET`.

Reply:
Oh, let me give that a shot. The only problem is that those are macros so I can't do the math symbolically — I'd have to hardcode values into the GDB script.

Reply:
Just force-pushed this. PTAL!

Reply:
Hardcoding is fine, thanks, I'll take a look.